libp2p/stream.go

```diff
@@ -0,0 +1,62 @@
+package libp2p
+
+import (
+	"bytes"
+	"io"
+	"time"
+
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rlp"
+	net "github.com/libp2p/go-libp2p-net"
+)
+
+func NewStream(s net.Stream, r, w time.Duration) Stream {
+	return Stream{
+		s:            s,
+		rlp:          rlp.NewStream(s, 0),
+		readTimeout:  r,
+		writeTimeout: w,
+	}
+}
+
+type Stream struct {
```

Comment:
`Stream` is super generic while this construct is producing rlp messages. It would be great if we can name somehow what we do here. `RLPTransformStream`?